woocommerce/woocommerce
# Update inbox notes to display localized strings when locale changed

### All Submissions:






### Changes proposed in this Pull Request:



Closes #33957.

This PR update inbox notes to display localized strings when a user changes the locale.

- Introduce `woocommerce_get_note_from_db` filter in `get_notes` method to modify the note before it is returned
- Add `woocommerce_get_note_from_db` filter in Events.php and RemoteInboxNotificationsEngine.php

We still need to update some notes that have no`get_note` method or notes in [plugins/woocommerce/includes/admin/notes/](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/includes/admin/notes) to support this i18n feature. I'll create follow-up issues for them.

### How to test the changes in this Pull Request:

**Test admin notes**

1. Use a fresh site
2. Go to `WooCommerce > Home`
3. Skip OBW
4. Observe that `Connect to WooCommerce.com` note is displayed
5. Change the "Site Language" to `Español` on the **Setting**  page
6. Go to `Dashboard > Updates`
7. Scroll to page bottom and make sure "Translations (Traducciones)" are all up to updated 
8. Go to `WooCommerce > Home`
9. Observe that `Connect to WooCommerce.com` note is translated to `Conectar con WooCommerce.com`.

**Test remote inbox notes** 

1. Use a fresh site
2. Change [this URL](https://github.com/woocommerce/woocommerce/blob/d8dd9853b60d6bd0c1eb0fe5426c56698bbec07b/plugins/woocommerce/src/Admin/RemoteInboxNotifications/DataSourcePoller.php#L18) to `https://gist.githubusercontent.com/chihsuan/e2aff67165428d2fd749b68cb7c75e74/raw/50a5adf7250240c6e667966b098d0001ef6c8faa/test_remote_inbox_specs`
4. Delete `_transient_woocommerce_admin_remote_inbox_notifications_specs` if it exists
5. Go to WooCommerce > Home
6. Skip OBW
7. Observer that `We want to know what matters most to you` note is displayed.
8. Change the "Site Language" to `Español` on the **Setting**  page
9. Go to `WooCommerce > Home`
10. Observer that `We want to know what matters most to you` note  is translated to `Queremos saber qué es lo que más te importa`.




### Other information:




### FOR PR REVIEWER ONLY:
